server/server/Controllers/adminCrud.js

```diff
@@ -0,0 +1,104 @@
+const itemsModel  = require("../models/Item");
+
+// Admin get item by id
+const getItemById = async (req, res) => {
+    const { id } = req.params;
+    try {
+        const item = await itemsModel.findById(id);
+        if (!item) {
+            // If the item is not found
+            return res.status(404).json({ message: "Item not found" });
+        }
+        // If the item is found
+        res.json(item);
+    } catch (err) {
+        // If there's an error
+        res.status(500).json({ message: err.message });
+    }
+};
+
+
+
+// admin add new item
+const addNewItem = async (req, res) => {
+  try {
+    const { title, price, itemPicture, category, availableQuantity } =
+      req.body;
+
+    const newItem = itemsModel({
+      title, 
+      price, 
+      itemPicture, 
+      category, 
+      availableQuantity,
+    });
+
+    await newItem.save();
+
+    res
+      .status(201)
+      .json({ message: "Item added successfully", newItem });
+  } catch (error) {
+    console.error("Error adding item:", error);
+    res.status(500).json({ message: error.message });
+  }
+};
+
+
+
+  // admin update item
+  const updateItem = async (req, res) => {
+    try {
+       const { id } = req.params; 
+       const { title, price, itemPicture, category, availableQuantity } = req.body; 
+       
+       const updatedItem = await itemsModel.findByIdAndUpdate(
+         id,
+         {
+           title,
+           price,
+           itemPicture,
+           category,
+           availableQuantity,
+         },
+         );
```

Comment:
also here we can pass a description.
in the `findByIdAndUpdate` function, for **validation rules** of schema to apply to the new changes and **return the updated** item we need to pass an object as a 3rd argument for the function like the suggestion

```suggestion
       const { title, price, description, itemPicture, category, availableQuantity } = req.body; 
       
       const updatedItem = await itemsModel.findByIdAndUpdate(
         id,
         {
           title,
           price,
           description,
           itemPicture,
           category,
           availableQuantity,
         },
         {
         runValidators: true,
         new: true,
         },
         );
```